### ISSUE TITLE
NodeModuleCopyHelper excluding certain top-level files leads to broken binaries

### ISSUE THREAD

**Issue by ahmed-at-mabl** [electron-userland/electron-builder#7795]
[In this line of code](https://github.com/electron-userland/electron-builder/blob/562870e92dea50b2d20e4fd8519ac8f5293b6336/packages/app-builder-lib/src/util/NodeModuleCopyHelper.ts#L16), "test.js" is added to this list of `topLevelExcludedFiles` that are not copied over when building the binary. These files are intended to be READMEs, test files etc. that can be safely removed from production binaries, but this assumption doesn't always hold, and it doesn't seem possible to overwrite `topLevelExcludedFiles` anywhere.

For example, [Playwright 1.38.0](https://www.npmjs.com/package/playwright/v/1.38.0) added a top-level [test.js file](https://github.com/microsoft/playwright/blob/d6ec1ae3994f127e38b866a231a34efc6a4cac0d/packages/playwright/test.js#L1) which cannot be safely excluded. Without it, when you attempt to open the built binary, it fails to load with message

```
Error: Cannot find module '/Users/.../<appName>.app/Contents/Resources/app/node_modules/playwright/test.js'
at createEsmNotFoundErr (node:internal/modules/cjs/loader:990:15)
```

I confirmed this was the cause of the above behavior by modifying the `node_modules/app-builder/lib/out/util/NodeModuleCopyHelper.js` file myself and removing `test.js` from the list of excluded files, then running the build command, and seeing that the built app worked.

This is with Node 16, npm v8. Tested only on a Mac but I assume the behavior is replicable with other OS'es.

There does not seem to be a workaround by modifying the `files` argument: adding `node_modules/**/playwright/test.js` to the `files` globs still results in the file not being copied over, which is what the code seems to suggest would occur.

In the short term, maybe `test.js` is not a filename that can be assumed to be safely removable, so it could be removed from that list. It seems like the ideal fix is that these top-level excluded files are pre-emptible with the passed in `files` argument, rather than being hardcoded in and apparently impossible to override.

**Comment by arvindell** [comment]
I have this same issue myself; had to extract the app.asar contents to figure out what was going wrong. `test.js` is an essential file in this case.

@ahmed-at-mabl Did you find any workaround in the end?

**Comment by mmaietta** [comment]
So I'm not sure why `test.js` is explicitly listed as something that should be excluded, as it seems to have existed that way for years. I'm happy to have it "allowed" as there's no code comment as to why it's excluded (maybe because it when-compiled could contain keys/secrets?)

**Comment by ahmed-at-mabl** [comment]
It would be a very appreciated change!

**Comment by mmaietta** [comment]
Released in 24.9.4

**Comment by arvindell** [comment]
@mmaietta Awesome, thanks!
